src/host/mongomerkle.rs

```diff
@@ -395,11 +334,11 @@ impl<const DEPTH: usize> MerkleTree<[u8; 32], DEPTH> for MongoMerkle<DEPTH> {
     type Root = [u8; 32];
     type Node = MerkleRecord;
 
-    fn construct(addr: Self::Id, root: Self::Root) -> Self {
+    fn construct(addr: Self::Id, root: Self::Root, db: Option<Rc<RefCell<dyn TreeDB>>>) -> Self {
```

Comment:
Wrong box, should be Option<Rc<RefCell<dyn TreeDB>>>.
Compile error when run cargo test.